src/openlifu/virtual_fit.py

```diff
@@ -329,17 +329,19 @@ def progress_callback(progress_percent : int, step_description : str): # noqa: A
     points_asl = np.zeros((num_search_points,3), dtype=float) # search grid points in ASL coordinates
     normals_asl = np.zeros((num_search_points,3), dtype=float) # normal vector of the plane that is fitted at each point, in ASL coordinates
 
+    # Pre-build plane fitting grid that will be used repeatedly in the search loop below.
+    # This grid lives in the spherical coordinate basis theta-phi plane at each point.
+    # Below we will project it onto the skin surface for each search point.
+    dtheta_sequence = np.arange(-planefit_dyaw_extent, planefit_dyaw_extent + planefit_dyaw_step, planefit_dyaw_step)
+    dphi_sequence = np.arange(-planefit_dpitch_extent, planefit_dpitch_extent + planefit_dpitch_step, planefit_dpitch_step)
+    dtheta_grid, dphi_grid = np.meshgrid(dtheta_sequence, dphi_sequence, indexing='ij')
```

Comment:
the construction of this point grid was previously inside the `for i in range(num_search_points):` just below, and I moved it out because it is the same grid each time.